circuit/src/arrays.nr

```diff
@@ -0,0 +1,18 @@
+pub fn sub_array_equals<T, N, M>(subarray: [T; N], array: [T; M], start_index: Field) -> bool {
+    let mut result = true;
```

Comment:
I would also return false if `start_index + M > N` rather then throw `Index out of bounds`